server/datastore/mysql/app_configs.go

```diff
@@ -23,6 +29,32 @@ func (d *Datastore) AppConfig() (*kolide.AppConfig, error) {
 	return info, nil
 }
 
+func (d *Datastore) ManageHostExpiryEvent(hostExpiryEnabled bool, hostExpiryWindow int) error {
+	rows, err := d.db.Query("SELECT @@event_scheduler")
+	if err != nil {
+		return err
+	}
+	if !rows.Next() {
+		return errors.New("Error detecting event scheduler status.")
+	}
+	var value string
+	if err := rows.Scan(&value); err != nil {
+		return err
+	}
+	if value != "ON" {
+		return errors.New("Event Scheduler must be enabled for configuring Host Expiry.")
+	}
+
+	if !hostExpiryEnabled {
+		_, err := d.db.Exec(fmt.Sprintf("DROP EVENT IF EXISTS %s", HostExpiryEvent))
```

Comment:
We avoid using string formatting to craft SQL queries in order to protect against SQL injection attacks. Please use parameters as you can see done in https://github.com/kolide/fleet/blob/master/server/datastore/mysql/app_configs.go#L27-L114.